src/Ocelot/Requester/HttpClientWrapper.cs

```diff
@@ -11,9 +11,12 @@ public class HttpClientWrapper : IHttpClient
     {
         public HttpClient Client { get; }
 
-        public HttpClientWrapper(HttpClient client)
+        public DelegatingHandler ClientMainHandler { get; }
+
+        public HttpClientWrapper(HttpClient client, DelegatingHandler clientMainHandler)
```

Comment:
Why not making clientMainHandler default null?

Reply:
Or define new overloaded method version. 😉